### PR TITLE
Task-46549: Implement mobile view for news detail

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
Prior to change, news details was created in one component. After this change, in order to make code more maintainable and to implement mobile view, we refactor the code by splitting into components specific to mobile view, others specific to desktop view and other common components to mobile and desktop views.

### CODE REVIEW

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 1]
```diff
@@ -203,81 +87,18 @@ export default {
...
     },
   },
   created() {
+    this.$root.$on('delete-news', this.deleteConfirmDialog);
+    this.$root.$on('edit-news', this.editLink);
     if (!this.news || !this.news.spaceId) {
       this.$newsServices.getNewsById(this.newsId)
         .then(news => {
           this.spaceId = news.spaceId;
+          this.getSpaceById(this.spaceId);
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue:101]: you can put them before the if block IMHO

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 2]
```diff
@@ -1,153 +1,37 @@
...
-    <exo-news-details-action-menu
-      v-if="showEditButton"
+    <exo-news-details-toolbar
+      v-if="!isMobile"
+      :news="news"
+      :news-id="newsId"
+      :activity-id="activityId"
+      :show-edit-button="showEditButton"
+      :show-delete-button="showDeleteButton"
+      :show-pin-button="showPinButton" />
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue]:
```suggestion
      :show-publish-button="showPublishButton" />
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 3]
```diff
@@ -1,153 +1,37 @@
...
-      @edit="editLink" />
-    <div class="newsDetailsIcons">
-      <exo-news-pin
-        v-if="showPinButton"
-        :news-id="newsId"
-        :news-pinned="news.pinned"
-        :news-archived="archivedNews"
-        :news-title="newsTitle" />
-    </div>
+      :show-pin-button="showPinButton" />
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue]:
```suggestion
      :show-publish-button="showPublishButton" />
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 4]
```diff
@@ -380,23 +171,6 @@ export default {
...
-        if (href !== location.host && !href.startsWith(internal)) {
-          links[i].setAttribute('target', '_blank');
-          links[i].setAttribute('rel', 'noopener noreferrer');
-        }
-      }
-      return docElement.innerHTML;
-    },
     postNews(schedulePostDate, postArticleMode, publish) {
       this.news.timeZoneId = USER_TIMEZONE_ID;
       this.news.pinned = publish;
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue:176]:
```suggestion
      this.news.published= publish;
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 5]
```diff
@@ -1,45 +1,59 @@
...
+          <v-list-item-title>
+            {{ $t('news.details.header.menu.resume') }}
+          </v-list-item-title>
+        </v-list-item>
+        <v-list-item v-if="showDeleteButton" @click="$root.$emit('delete-news')">
+          <v-list-item-title>
+            {{ $t('news.details.header.menu.delete') }}
+          </v-list-item-title>
+        </v-list-item>
+        <v-list-item v-if="isMobile && showPinButton" @click="confirmAction">
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
        <v-list-item v-if="isMobile && showPublishButton" @click="confirmAction">
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 6]
```diff
@@ -1,45 +1,59 @@
...
+        </v-list-item>
+      </v-list>
+    </v-menu>
+    <exo-confirm-dialog
+      ref="publishConfirmDialog"
+      :title="confirmDialogTitle"
+      :message="confirmDialogMessage"
+      :ok-label="$t('news.broadcast.btn.confirm')"
+      :cancel-label="$t('news.broadcast.btn.cancel')"
+      @ok="updatePinnedField" />
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
      @ok="updatePublishedField" />
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 7]
```diff
@@ -51,6 +65,16 @@ export default {
       required: false,
       default: null
     },
+    newsPinned: {
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
    newsPublished: {
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 8]
```diff
@@ -71,15 +95,90 @@ export default {
       required: false,
       default: false
     },
+    showPinButton: {
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
    showPublishButton: {
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 9]
```diff
@@ -71,15 +95,90 @@ export default {
...
+      type: Boolean,
+      required: false,
+      default: false
+    },
   },
   data: () => ({
     actionMenu: null,
+    messagePin: '',
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
    publishMessage: '',
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 10]
```diff
@@ -71,15 +95,90 @@ export default {
...
+    successPin: true,
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
    publishSuccess: true,
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 11]
```diff
@@ -71,15 +95,90 @@ export default {
...
   }),
+  computed: {
+    isMobile() {
+      return this.$vuetify.breakpoint.name === 'xs' || this.$vuetify.breakpoint.name === 'sm';
+    },
+    confirmDialogTitle() {
+      return this.newsPinned && this.$t('news.unbroadcast.action') || this.$t('news.broadcast.action');
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
      return this.newsPublished && this.$t('news.unpublisht.action') || this.$t('news.publish.action');
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 12]
```diff
@@ -71,15 +95,90 @@ export default {
...
+    },
+    confirmDialogMessage() {
+      return this.newsPinned && this.$t('news.unbroadcast.confirm', {0: this.news.title}) || this.$t('news.broadcast.confirm');
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
      return this.newsPublished && this.$t('news.unpublish.confirm', {0: this.news.title}) || this.$t('news.publish.confirm');
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue [hunk 13]
```diff
@@ -71,15 +95,90 @@ export default {
...
+    },
+    publishLabel() {
+      return this.newsPinned ? this.$t('news.details.header.menu.unpublish'): this.$t('news.details.header.menu.publish');
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsActionMenu.vue]:
```suggestion
      return this.newsPublished ? this.$t('news.details.header.menu.unpublish'): this.$t('news.details.header.menu.publish');
```

File: webapp/src/main/webapp/news-details/components/ExoNewsPin.vue [hunk 14]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div id="pinNewsActivity">
+  <div id="pinNewsActivity" class="newsDetailsIcons">
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsPin.vue]:
```suggestion
  <div id="publishNewsActivity" class="newsDetailsIcons">
```

File: webapp/src/main/webapp/news-details/components/ExoNewsPin.vue [hunk 15]
```diff
@@ -1,5 +1,5 @@
 <template>
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsPin.vue:1]: Rename the file to ExoNewsPublish.vue

File: webapp/src/main/webapp/skin/less/news.less [hunk 16]
```diff
@@ -29,9 +29,20 @@
...
     }
   }
+  #newsDetails {
+    #newsFullDetails {
+      .newsContent {
+        background-color: @baseBackground;
+      }
+    }
+  }
   #newsDetails .newsDetailsActionMenu {
```
Comment (azayati) [inline webapp/src/main/webapp/skin/less/news.less]: put .newsDetailsActionMenu class under #newsDetails {

File: webapp/src/main/webapp/skin/less/news.less [hunk 17]
```diff
@@ -29,9 +29,20 @@
...
     margin: 20px 20px 13px 0;
   }
+  #newsDetails .newsDetailsMenuBtn {
```
Comment (azayati) [inline webapp/src/main/webapp/skin/less/news.less]: put .newsDetailsMenuBtnclass under #newsDetails {

File: webapp/src/main/webapp/skin/less/news.less [hunk 18]
```diff
@@ -3856,6 +3863,14 @@ p.caption-title:after {
     display: none;
   }
 }
+@media (max-width: 959px) {
+  .VuetifyApp #newsDetail {
+    padding: 0;
+  }
+  .VuetifyApp #newsDetails .newsDetailsActionMenu {
```
Comment (azayati) [inline webapp/src/main/webapp/skin/less/news.less]: put .newsDetailsActionMenu { class under .VuetifyApp #newsDetail { class

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue (hunk 1 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue:101]: we get space Id when we call getNewsById

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue (hunk 4 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue:176]: In the back end the name of field is pinned.

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 19]
```diff
@@ -294,13 +115,10 @@ export default {
         });
     } else {
       this.spaceId = this.news.spaceId;
+      this.getSpaceById(this.spaceId );
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue:118]:
```suggestion
      this.getSpaceById(this.spaceId);
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**